web/pdf_page_view.js

```diff
@@ -104,42 +183,275 @@ const DEFAULT_LAYER_PROPERTIES =
 
 const LAYERS_ORDER = new Map([
   ["canvasWrapper", 0],
-  ["textLayer", 1],
-  ["annotationLayer", 2],
-  ["annotationEditorLayer", 3],
-  ["xfaLayer", 3],
+  ["detailLayer", 1],
+  ["textLayer", 2],
+  ["annotationLayer", 3],
+  ["annotationEditorLayer", 4],
+  ["xfaLayer", 4],
 ]);
 
+class PDFPageViewBase {
```

Comment:
Given the size and scope of this patch, it feels quite difficult to review as-is and catching any possible mistakes is going to be really difficult.

Hence I'll "insist" that this PR should have *separate commits* for easier reviewing:
 - A first commit that only introduces the `PDFPageViewBase` class.
 - A second commit that adds the actual "partial rendering" functionality.

Reply:
Sounds good, I'll split it 👍